collect_app/src/main/java/org/odk/collect/android/activities/FormMapActivity.java

```diff
@@ -104,10 +104,11 @@
         List<Form> forms = dao.getFormsFromCursor(dao.getFormsCursor(getIntent().getData()));
         if (forms.size() == 1) {
             form = forms.get(0);
-        } else {
+        } else if (viewModelFactory == null) {
```

Comment:
Why add this condition? I think we want to exit in any other case. For example, it could be possible for no form to match the id.

Reply:
`viewModelFactory` should only be `null` in tests and in that case it will always be `null`.

Reply:
Oh, I see. It's `null` in the NON-test case and so you're avoiding this branch for tests. But can't you instead make sure the tests always have exactly one form so the `else` branch doesn't get hit?